devops/scripts/publish_register_bundle.sh

```diff
@@ -63,12 +64,15 @@ while [ "$1" != "" ]; do
         usage
         ;;
     esac
+    if [[ -z "$2" ]]; then
+      # if no more args then stop processing
+      break
+    fi
     shift # remove the current value for `$1` and use the next
 done
 
 if [[ -z ${access_token} ]]; then
     echo -e "WARNING!!! No Azure access token provided. Automatic bundle registration not possible. Use the script output to self-register. See documentation for more details.\n"
-    usage
 fi
 
 if [[ -z ${tre_url} ]]; then
```

Comment:
line 80  "usage" in check on tre_url is irrelevant if an access_token is not provided and causes the script to fail.

Suggest moving if statement on like 78 down to after line 101 adding in the following at line 76:
 elif [[ -n  ${access_token} ]]; then
echo -e "No TRE URL provided\n"
      usage
    fi

Reply:
I've updated to allow no url if there's no access_token

Thanks!

Reply:
Looks good to me and runs locally as expected